packages/core/url/src/index.ts

```diff
@@ -119,10 +120,19 @@ export function getPublicUrl(url_prefix: string = '', requestOptions: RequestOpt
     if (!isHost(host)) {
       throw new Error('invalid host');
     }
-    const protoHeader =
+
+    const protoHeader: string =
       process.env.VERDACCIO_FORWARDED_PROTO?.toLocaleLowerCase() ??
       HEADERS.FORWARDED_PROTO.toLowerCase();
-    const protocol = getWebProtocol(requestOptions.headers[protoHeader], requestOptions.protocol);
+    const forwardedProtocolHeaderValue = requestOptions.headers[protoHeader];
+
+    if (Array.isArray(forwardedProtocolHeaderValue)) {
+      // This really should never happen - only cookies are allowed to have
+      // multiple values.
+      throw new Error('invalid protocol header value');
```

Comment:
I'd consider this at least a bit bigger than a `chore` between minor and patch.

Reply:
You're right. It should have been patch I think

Reply:
Now I kind of want to split this PR up into two. First one where I just make this change, and then one where I improve the types. Do you want me to do that?

Reply:
I would recommend remove this part in order do not block the types part so I can give it some additional thoughts, because I remember this part might be an array, I barely remember